배수연/5월 1주차/BOJ_G3_1256.java

```diff
@@ -0,0 +1,77 @@
+package boj;
+
+import java.util.Arrays;
+import java.util.Scanner;
+
+/**
+ * 전체 경우의 수 : (중복조합) nHm = n+m-1Cm 
+ * nCm = n-1Cm + n-1Cm-1
+ * n+mCm = n+m-1Cm + n+m-1Cm-1
+ * nHm 	= n+m-1Cm
+ * 		= n+m-2Cm + n+m-2Cm-1
+ *		= n-1Hm + nHm-1
+ * 
+ * 맨 앞 한 자리 고정한 경우의 수를 K와 비교
+ * K보다 작으면 맨 앞 a로 고정, 크면 z로 시작
+ * 반복
+ */
+
+public class BOJ_G3_1256 {
+	static int N, M, K;
+	static long[][] DP;
+	static String ans;
+
+	public static void main(String[] args) {
+		Scanner sc = new Scanner(System.in);
+		N = sc.nextInt();
+		M = sc.nextInt();
+		K = sc.nextInt();
+		DP = new long[N+1][M+1];
+		ans = "";
+		
+		// 모든 경우의 수 구해서 DP에 넣기
+		getDP();
+		
+		if(K>DP[N][M]) ans = "-1";
+		else find(N, M, K-1);
+		System.out.println(ans);
+	}
+
+	private static void getDP() {
+		for(int n = 0 ; n<=N ; n++) {
+			Arrays.fill(DP[n], 1);
+		}
+		for(int n = 1 ; n<=N ; n++) {
+			for(int m = 1 ; m<=M ; m++) {
```

Comment:
모든 테이블을 `1`로 채워둘 필요는 없을 것 같아요!!
```suggestion
		Arrays.fill(DP[0], 1);
		for(int n = 1 ; n<=N ; n++) {
			d[n][0] = 1;
			for(int m = 1 ; m<=M ; m++) {
```